### PR TITLE
Enhance error handling/logging for network and message errors

### PR DESCRIPTION
**Pull request by Apollon77**
This PR just catches some error and make sure they are visible and provide more details.
On network level the "catching on the lowest level" should also make sure it is announced as good as possible

### CODE REVIEW

File: src/matter/common/ExchangeManager.ts [hunk 1]
```diff
@@ -53,7 +53,7 @@ export class ExchangeManager<ContextT> {
     addNetInterface(netInterface: NetInterface) {
         this.netListeners.push(netInterface.onData((socket, data) => {
             this.onMessage(socket, data)
-                .catch(error => logger.error(error));
+                .catch(error => logger.error(error.stack));
```
Comment (mfucci) [inline src/matter/common/ExchangeManager.ts:56]: Maybe this should be modified in Logger.ts to always display the stack on .error() when the argument is of Error class?
I don't see a reason why not to always display the stack...

File: src/net/UdpMulticastServer.ts [hunk 2]
```diff
@@ -52,7 +55,7 @@ export class UdpMulticastServer {
             const { ips } = this.network.getIpMac(netInterface) ?? { ips: [] };
             await Promise.all(ips.map(async ip => {
                 const iPv4 = isIPv4(ip);
-                await (await this.broadcastChannels.get(netInterface, iPv4)).send(iPv4 ? this.broadcastAddressIpv4 : this.broadcastAddressIpv6, this.broadcastPort, message);
+                await (await this.broadcastChannels.get(netInterface, iPv4)).send(iPv4 ? this.broadcastAddressIpv4 : this.broadcastAddressIpv6, this.broadcastPort, message).catch(err => logger.debug(`${netInterface}: ${err}`));
```
Comment (mfucci) [inline src/net/UdpMulticastServer.ts:58]: This is an async function so maybe better to write this using try / catch;
```
try {
     const { ips } = this.network.getIpMac(netInterface) ?? { ips: [] };
            await Promise.all(ips.map(async ip => {
                const iPv4 = isIPv4(ip);
                await (await this.broadcastChannels.get(netInterface, iPv4)).send(iPv4 ? this.broadcastAddressIpv4 : this.broadcastAddressIpv6, this.broadcastPort, message));
              }));
        }));
} catch (error) {
    logger.debug(`${netInterface}: ${error}`);
}
```

Comment (mfucci) [inline src/net/UdpMulticastServer.ts:58]: logger.warn ?
I think this match the definition of a warning: error but with no expected functional impact

**Review by mfucci — CHANGES_REQUESTED** [review]

File: src/net/UdpMulticastServer.ts (hunk 2 above)
Comment (Apollon77) [reply src/net/UdpMulticastServer.ts:58]: Hm ... for Me warn is something where the user should react on but here he can not even do anything. That's why for me maximal info ;-)

File: src/matter/common/ExchangeManager.ts (hunk 1 above)
Comment (Apollon77) [reply src/matter/common/ExchangeManager.ts:56]: Ok

File: src/net/UdpMulticastServer.ts (hunk 2 above)
Comment (Apollon77) [reply src/net/UdpMulticastServer.ts:58]: I adjusted it to info.

**Comment by Apollon77** [comment]
@mfucci Ready for re-review

File: src/net/UdpMulticastServer.ts [hunk 3]
```diff
@@ -52,7 +55,11 @@ export class UdpMulticastServer {
             const { ips } = this.network.getIpMac(netInterface) ?? { ips: [] };
             await Promise.all(ips.map(async ip => {
                 const iPv4 = isIPv4(ip);
-                await (await this.broadcastChannels.get(netInterface, iPv4)).send(iPv4 ? this.broadcastAddressIpv4 : this.broadcastAddressIpv6, this.broadcastPort, message);
+                try {
+                    await (await this.broadcastChannels.get(netInterface, iPv4)).send(iPv4 ? this.broadcastAddressIpv4 : this.broadcastAddressIpv6, this.broadcastPort, message);
+                } catch (err) {
+                    logger.debug(`${netInterface}: ${(err as Error).message}`);
```
Comment (mfucci) [inline src/net/UdpMulticastServer.ts:61]: logger.info?

**Review by mfucci — DISMISSED** [review]

File: src/net/UdpMulticastServer.ts (hunk 3 above)
Comment (Apollon77) [reply src/net/UdpMulticastServer.ts]: adjusted. @mfucci Re-Review please ;-)) (or change that setting? ;-) )

**Review by mfucci — APPROVED** [review]

**Pull request merged**